File: P1/ej4/filters/filter_manager.py
from .filter_chain import FilterChain

class FilterManager:
    def __init__(self, target):
        self.filter_chain = FilterChain()
        self.filter_chain.set_target(target)
    
    def add_filter(self, filter_obj):
        self.filter_chain.add_filter(filter_obj)
    
    def process_request(self, request):
        errors = self.filter_chain.execute(request)
        if errors:
            return {'success': False, 'errors': errors}
        return self.filter_chain.target.process(request)

File: P2/ejercicio1/lib/filters/filter_manager.dart
import 'base.dart';
import 'filter_chain.dart';

class FilterManager {
  FilterChain filterChain;
  List<String> emails_used = [];

  FilterManager(dynamic target) : filterChain = FilterChain() {
    filterChain.setTarget(target);
  }

  void addFilter(Filter filterObj) {
    filterChain.addFilter(filterObj);
  }

  Map<String, dynamic> processRequest(Map<String, String> request) {
    List<String> errors = filterChain.execute(request);
    if (errors.isNotEmpty) {
      return {'success': false, 'errors': errors};
    }
    if (emails_used.contains(request['email'])) {
      return {'success': false, 'errors': ['Este correo ya está en uso']};
    } else {
      emails_used.add(request['email'] as String);
      return filterChain.target.process(request);
    }
  }
}
